wordpress-mobile/WordPress-iOS
# Stats: Add Subscriber Chart

Addresses https://github.com/wordpress-mobile/WordPress-iOS/issues/23044

I looked to reuse as much of the Views & Visitors card as possible but found that hiding its segmented controls and various labels was difficult. I re-used `StatsLineChartView` but made a new cell, `StatsSubscribersChartCell` and a chart configuration object, `StatsSubscribersLineChart`.

# Screenshot

| <img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1898325/34ab9d91-a6ee-422b-abf7-20e869e6a93b" width="350" /> |
| - |


# To test

1. Log into a WP.com account
2. Enable Stats Traffic and Subscribers tab feature flag
3. Go to Stats > Subscribers tab on a site and verify the chart shows the same data as on WP.com's stats
4. Try different scenarios such as no subscribers (new site) and a lot of subscribers 
5. Try pull-to-refresh, connectivity issues, etc.
6. Compare with design: [IqhXWz3Iir7RMb5XH5gGfZ-fi-883%3A4006](https://href.li/?https://www.figma.com/file/IqhXWz3Iir7RMb5XH5gGfZ/?node-id=883%3A4006)
7. Compare with specs: https://github.com/Automattic/wordpress-mobile/issues/47


## Known issues

Some updates I'd like to do in follow-up PRs

- The line chart often looks flat, and small fluctuations in subscribers are not easy to see. The problem is that the y-axis minimum is always set to zero when it should be roughly (or exactly) to the minimum value of the chart. 
- The chart card loading ghost view doesn't resemble the chart
- The chart tooltip is not set up yet
- Reduce x-axis gradation
- ~~Scenarios where there are less then 30 days worth of data returned by `/sites/<site id>/stats/subscribers` such as a new or recently created site (💥 crashes now on new sites which return one day of data)~~ fixed in 9cc7ba2532e92788340987f6e70530631e9f3148

## Regression Notes
1. Potential unintended areas of impact

Potentially this P2 could affect the Subscriber email list

9. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

10. What automated tests I added (or what prevented me from doing so)

I added a new test to verify the chart card loads correctly

PR submission checklist:


Testing checklist:
